src/openslide-vendor-dicom.c

```diff
@@ -0,0 +1,932 @@
+  }
+
+  // true to remove this file
+  const char *this_slide_id;
+  return !get_tag_str(f->metadata, SeriesInstanceUID, 0, &this_slide_id) ||
+         !g_str_equal(slide_id, this_slide_id);
+}
+
+static int add_level_array(gpointer key G_GNUC_UNUSED, 
+                           gpointer value, 
+                           gpointer user_data) {
+  struct dicom_level *l = (struct dicom_level *) value;
+  GPtrArray *level_array = (GPtrArray *) user_data;
+
+  g_ptr_array_add(level_array, l);
+
+  return true;
+}
+
+static void print_file_from_hash(gpointer key G_GNUC_UNUSED, 
+                                 gpointer value, 
+                                 gpointer user_data G_GNUC_UNUSED) {
+  struct dicom_file *f = (struct dicom_file *) value;
+
+  print_file(f);
+}
+
+static void print_level_from_hash(gpointer key G_GNUC_UNUSED, 
+                                  gpointer value, 
+                                  gpointer user_data G_GNUC_UNUSED) {
+  struct dicom_level *l = (struct dicom_level *) value;
+
+  print_level(l);
+}
+
+static int find_associated(gpointer key G_GNUC_UNUSED, 
+                            gpointer value, 
+                            gpointer user_data) {
+  struct dicom_file *f = (struct dicom_file *) value;
+  openslide_t *osr = (openslide_t *) user_data; 
+
+  struct associated *a = associated_new(f);
+  if (a) {
+    // map dicom associated image names to openslide names
+    char *openslide_name = NULL;
+    if (g_str_equal(a->name, "LABEL")) {
+      openslide_name = "label";
+    }
+    else if (g_str_equal(a->name, "OVERVIEW")) {
+      openslide_name = "macro";
+    }
+
+    if (openslide_name) {
+      g_hash_table_insert(osr->associated_images, 
+                          g_strdup(openslide_name), 
+                          a);
+      return true;
+    }
+  }
+
+  return false;
+}
+
+static gint compare_level_width(const void *a, const void *b) {
+  const struct dicom_level *aa = *((const struct dicom_level **) a);
+  const struct dicom_level *bb = *((const struct dicom_level **) b);
+
+  return bb->image_width - aa->image_width;
+}
+
+static void make_grid(gpointer key G_GNUC_UNUSED, 
+                      gpointer value, 
+                      gpointer user_data) {
+  struct dicom_level *l = (struct dicom_level *) value;
+  openslide_t *osr = (openslide_t *) user_data; 
+
+  l->grid = _openslide_grid_create_simple(osr,
+                                          l->tiles_across, 
+                                          l->tiles_down,
+                                          l->tile_w, 
+                                          l->tile_h,
+                                          read_tile);
+}
+
+static void add_properties(openslide_t *osr, const struct dicom_level *l) {
+  // why not
+  double mmpp = l->image_width;
+  g_hash_table_insert(osr->properties,
+                      g_strdup(OPENSLIDE_PROPERTY_NAME_MPP_X),
+                      _openslide_format_double(mmpp * 1000));
```

Comment:
Wait, so we're saying the number of microns per pixel is exactly 1000x the number of pixels in X?  That doesn't seem right.

Reply:
It's just a placeholder until we add full metadata support (coming in phase 2).

Reply:
Let's not intentionally add incorrect metadata.  We can drop this for now and re-add in phase 2.

Reply:
It's just a placeholder :( it'll be replaced within a couple of weeks.